StyleCop.Analyzers/StyleCop.Analyzers/Helpers/IndentationHelper.cs

```diff
@@ -92,19 +91,23 @@ public static SyntaxTrivia GenerateWhitespaceTrivia(IndentationOptions indentati
 
         private static int GetIndentationSteps(IndentationOptions indentationOptions, SyntaxTriviaList leadingTrivia)
         {
-            var indentationString = string.Empty;
-            for (var i = leadingTrivia.Count - 1; (i >= 0) && leadingTrivia[i].IsKind(SyntaxKind.WhitespaceTrivia); i--)
+            SyntaxTriviaList.Reversed reversed = leadingTrivia.Reverse();
+            int indentationCount = 0;
+
+            foreach (SyntaxTrivia trivia in reversed)
             {
-                indentationString = string.Concat(leadingTrivia[i].ToFullString(), indentationString);
+                if (!trivia.IsKind(SyntaxKind.WhitespaceTrivia))
+                {
+                    break;
+                }
+
+                foreach (char c in trivia.ToFullString())
+                {
+                    indentationCount += c == '\t' ? indentationOptions.TabSize : 1;
+                }
             }
 
-            var indentationCount = indentationString.ToCharArray().Sum(c => IndentationAmount(c, indentationOptions));
             return indentationCount / indentationOptions.IndentationSize;
```

Comment:
`IndentationSize` should be already included as trivia itself in `leadingTrivia` list